lldb/source/Plugins/SymbolLocator/Debuginfod/SymbolLocatorDebuginfod.cpp

```diff
@@ -19,6 +20,43 @@ using namespace lldb_private;
 
 LLDB_PLUGIN_DEFINE(SymbolLocatorDebuginfod)
 
+enum SymbolLookupMode {
+  eLookupModeNever,
+  /*
+  eLookupModeAsync,
+  eLookupModeOnDemand,
+  */
+  eLookupModeAlways,
+};
+
+static constexpr OptionEnumValueElement g_debuginfod_symbol_lookup_mode[] = {
+    {
+        eLookupModeNever,
+        "never",
+        "Do not ever query DEBUGINFOD servers for symbols.",
+    },
+    /*
+    {
+        eLookupModeAsync,
+        "async",
+        "Only query for symbols asynchronously.",
+    },
```

Comment:
Do we want a boolean option for `debuginfod-async`? Async could be used for both `"on-demand"` or for `"always"` right? Of course if we have the symbols cached already, async would be done right away with no need for async, but when we start up a new debug session with all new system libraries, it would be great if async can be enabled. Or maybe async would _only_ apply to `"always"` since if the user types a command asking for symbols this should be a synchronous operation.